compiler/sigmatch.nim

```diff
@@ -1421,6 +1421,10 @@ proc prepareOperand(c: PContext; formal: PType; a: PNode): PNode =
     # {tyTypeDesc, tyExpr, tyStmt, tyProxy}:
     # a.typ == nil is valid
     result = a
+  elif formal.kind == tyVarargs and formal.sons[0].kind == tyExpr and
+      c.p.owner.kind notin {skProc, skMacro}:
```

Comment:
this is wrong. There is also skClosureIterator, skIterator, skConverter, skMethod; these all should be handled like skProc but I'm not sure why you want to exclude skTemplate here. Why is there a test for c.p.owner here?